async_simple/coro/Generator.h

```diff
@@ -0,0 +1,573 @@
+/*
+ * Copyright (c) 2023, Alibaba Group Holding Limited;
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ASYNC_SIMPLE_CORO_GENERATOR_H
+#define ASYNC_SIMPLE_CORO_GENERATOR_H
+
+#if __has_include(<generator>)
+
+#include <generator>
+
+#else
+
+#include <cassert>
+#include <concepts>
+#include <cstddef>
+#include <exception>
+#include <ranges>
+#include <type_traits>
+#include <utility>
+
+#include "async_simple/Common.h"
+#include "async_simple/coro/PromiseAllocator.h"
+#include "async_simple/experimental/coroutine.h"
+
+#endif  // __has_include(<generator>)
```

Comment:
```suggestion
#if __has_include(<generator>) && defined(__cpp_lib_generator)

#include <generator>

namespace async_simple::coro {
template<...>
using Generator = std::generator<...>;
}
namespace async_simple::ranges {
template<...>
using elements_of = std::ranges::elements_of<...>;
}

#else

#include <cassert>
#include <concepts>
#include <cstddef>
#include <exception>
#include <ranges>
#include <type_traits>
#include <utility>

#include "async_simple/Common.h"
#include "async_simple/coro/PromiseAllocator.h"
#include "async_simple/experimental/coroutine.h"

// the rest of the file
#endif // __has_include(<generator>)
```

感觉这样好一些吧，比较直观。这篇提案应该就引入了两个新声明。